src/Server/HTTP/HTTPServerConnection.cpp

```diff
@@ -29,6 +30,14 @@ void HTTPServerConnection::run()
             {
                 HTTPServerResponse response(session);
                 HTTPServerRequest request(context, response, session);
+                /// Check method (only supported methods will be processed)
```

Comment:
I don't think that we should handle it here. The issue is that default dynamic query handler [is created](https://github.com/ClickHouse/ClickHouse/blob/290a6d273eafcd2b7d7fad8624e7f2d7982a03bf/src/Server/HTTPHandlerFactory.cpp#L182-L184) with incorrect [filter](https://github.com/ClickHouse/ClickHouse/blob/414f470c79eb22b0ca47b82f11625cf80b0231aa/src/Server/HTTPHandlerFactory.h#L106-L114), so we choose this handler for custom http methods and do not fallback to a code that handles unknown/not implemented requests.